### PR TITLE
Update ARM template for IotHub to include IP Filter rules

### PR DESCRIPTION
**Pull request by rajeevmv**

### CODE REVIEW

File: schemas/2016-02-03/Microsoft.Devices.json [hunk 1]
```diff
@@ -241,6 +241,21 @@
             }
           ]
         },
+        "ipFilterRules": {
```
Comment (wenwu449) [inline schemas/2016-02-03/Microsoft.Devices.json]: do you have apiVersion change for this new property?

File: schemas/2016-02-03/Microsoft.Devices.json [hunk 2]
```diff
@@ -355,6 +370,32 @@
         }
       },
       "description": "The properties of the Storage Endpoint provided by the customer for file upload"
+    },
+    "IpFilterRule": {
+      "description": "IP filter Rule",
+      "type": "object",
+      "properties": {
+        "filterName": {
```
Comment (wenwu449) [inline schemas/2016-02-03/Microsoft.Devices.json]: please add expression support for all properties.

File: schemas/2016-02-03/Microsoft.Devices.json (hunk 1 above)
Comment (rajeevmv) [inline schemas/2016-02-03/Microsoft.Devices.json]: No. This change is backward-compatible.

File: schemas/2016-02-03/Microsoft.Devices.json (hunk 2 above)
Comment (rajeevmv) [inline schemas/2016-02-03/Microsoft.Devices.json]: I don't understand what it means to add expression support. Can you please explain?

Comment (wenwu449) [inline schemas/2016-02-03/Microsoft.Devices.json]: To support customer to use template expression for any properties, we need to have following reference: "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"

Comment (rajeevmv) [inline schemas/2016-02-03/Microsoft.Devices.json]: Done

File: schemas/2016-02-03/Microsoft.Devices.json [hunk 3]
```diff
@@ -355,6 +370,53 @@
...
+            {
+              "description": "The action desired - accept or reject.",
+              "enum": [ "Accept", "Reject" ]
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" 
+            }
+          ]
+        },
+        "ipMask": {
```
Comment (wenwu449) [inline schemas/2016-02-03/Microsoft.Devices.json]: "string" type property already support "expression" as expression itself is a string, the "oneOf" would make VS failed to match the property.
Please revert this ipMask property to last iteration.

Comment (rajeevmv) [inline schemas/2016-02-03/Microsoft.Devices.json]: done

**Pull request merged**